internal/nodes/avi_model_evh_nodes.go

```diff
@@ -486,6 +497,112 @@ func (v *AviEvhVsNode) GetNodeType() string {
 	return "VirtualServiceNode"
 }
 
+func (o *AviEvhVsNode) AddFQDNAliasesToHTTPPolicy(host string, hosts []string, key string) {
+
+	var hppMap *AviHostPathPortPoolPG
+	var redirectPorts *AviRedirectPort
+
+	// Find the hppMap and redirectPorts that matches the host
+	for _, policy := range o.HttpPolicyRefs {
+		for j := range policy.HppMap {
+			if utils.HasElem(policy.HppMap[j].Host, host) {
+				hppMap = &policy.HppMap[j]
+				break
+			}
+		}
+		for j := range policy.RedirectPorts {
+			if utils.HasElem(policy.RedirectPorts[j].Hosts, host) {
+				redirectPorts = &policy.RedirectPorts[j]
+				break
+			}
+		}
+		if utils.HasElem(policy.AviMarkers.Host, host) {
+			for _, host := range hosts {
+				if !utils.HasElem(policy.AviMarkers.Host, host) {
+					policy.AviMarkers.Host = append(policy.AviMarkers.Host, host)
+				}
+			}
+		}
+	}
+
+	// Update the hppMap with the hosts
+	if hppMap != nil {
+		for _, host := range hosts {
+			if !utils.HasElem(hppMap.Host, host) {
+				hppMap.Host = append(hppMap.Host, host)
+			}
+		}
+	}
+
+	// Update the redirectPorts with the hosts
+	if redirectPorts != nil {
+		for _, host := range hosts {
+			if !utils.HasElem(redirectPorts.Hosts, host) {
+				redirectPorts.Hosts = append(redirectPorts.Hosts, host)
+			}
+		}
+	}
+
+	utils.AviLog.Debugf("key: %s, msg: Added multiple FQDNs to HTTP policy for VS %s", key, o.Name)
+}
+
+func (o *AviEvhVsNode) RemoveFQDNAliasesFromHTTPPolicy(host string, hosts []string, key string) {
+
+	// Find the hppMap and redirectPorts that matches the host and remove the hosts
+	for _, policy := range o.HttpPolicyRefs {
+		for j := range policy.HppMap {
+			if utils.HasElem(policy.HppMap[j].Host, host) {
+				for _, host := range hosts {
+					policy.HppMap[j].Host = utils.Remove(policy.HppMap[j].Host, host)
+				}
+			}
+		}
+		for j := range policy.RedirectPorts {
+			if utils.HasElem(policy.RedirectPorts[j].Hosts, host) {
+				for _, host := range hosts {
+					policy.RedirectPorts[j].Hosts = utils.Remove(policy.RedirectPorts[j].Hosts, host)
+				}
+			}
+		}
+		if utils.HasElem(policy.AviMarkers.Host, host) {
+			for _, host := range hosts {
+				policy.AviMarkers.Host = utils.Remove(policy.AviMarkers.Host, host)
+			}
+		}
+	}
+
+	utils.AviLog.Debugf("key: %s, msg: Removed multiple FQDNs from HTTP policy for VS %s", key, o.Name)
+}
+
+func (o *AviEvhVsNode) AddFQDNsToModel(hosts []string, gsFqdn, key string) {
+	if len(o.VSVIPRefs) == 0 {
+		return
+	}
+	for _, host := range hosts {
+		if host != gsFqdn &&
+			!utils.HasElem(o.VSVIPRefs[0].FQDNs, host) {
+			o.VSVIPRefs[0].FQDNs = append(o.VSVIPRefs[0].FQDNs, host)
+		}
+	}
+	utils.AviLog.Debugf("key: %s, msg: Added multiple FQDNs to model for VS %s", key, o.Name)
```

Comment:
instead of printing the name of the VS can we print the `hosts`?

Reply:
I have added the hosts in the log message as per the comment.